types/underscore/index.d.ts

```diff
@@ -3436,36 +3452,27 @@ declare module _ {
         values(object: any): any[];
 
         /**
-         * Like map, but for objects. Transform the value of each property in turn.
-         * @param object The object to transform
-         * @param iteratee The function that transforms property values
-         * @param context The optional context (value of `this`) to bind to
-         * @return a new _.Dictionary of property values
-         */
-        mapObject<T, U>(object: _.Dictionary<T>, iteratee: (val: T, key: string, object: _.Dictionary<T>) => U, context?: any): _.Dictionary<U>;
-
-        /**
-         * Like map, but for objects. Transform the value of each property in turn.
-         * @param object The object to transform
-         * @param iteratee The function that tranforms property values
-         * @param context The optional context (value of `this`) to bind to
-         */
-        mapObject<T>(object: any, iteratee: (val: any, key: string, object: any) => T, context?: any): _.Dictionary<T>;
-
-        /**
-         * Like map, but for objects. Retrieves a property from each entry in the object, as if by _.property
-         * @param object The object to transform
-         * @param iteratee The property name to retrieve
-         * @param context The optional context (value of `this`) to bind to
+         * Like map, but for objects. Transform the value of each property in
+         * turn.
+         * @param object The object to transform.
+         * @param iteratee The iteratee to use to transform property values.
+         * @param context `this` object in `iteratee`, optional.
+         * @returns A new object with all of `object`'s property values
+         * transformed through `iteratee`.
          */
-        mapObject(object: any, iteratee: string, context?: any): _.Dictionary<any>;
+        mapObject<V extends object, I extends ObjectIteratee<V, any>>(
+            object: V,
+            iteratee: I,
+            context?: any
+        ): { [K in keyof V]: ObjectIterateeResult<I, V[K]> };
```

Comment:
I considered aliasing this object signature to a type like

```TS
type ObjectResult<V, I> = { [K in keyof V]: ObjectIterateeResult<I, V[K]> };
```

But then `getQuickInfoAtPosition` (probably) seems to wind up with results that look like this:

```TS
ObjectResult<Dictionary<StringRecord>, (element: StringRecord, key: string, dictionary: Dictionary<StringRecord>) => string>
```

Which doesn't seem as nice as results that lay out the properties of the resulting object.

I suppose, though, it might be worth renaming `ObjectIterateeResult` to `ObjectPropertyIterateeResult` or something since this type is probably the more accurate one to call `ObjectIterateeResult`.

Reply:
`ObjectIterateeResult` is gone, 🎉